python/ray/experimental/channel/cached_channel.py

```diff
@@ -0,0 +1,95 @@
+import uuid
+from typing import Any, Optional
+
+from ray.experimental.channel import ChannelContext
+from ray.experimental.channel.common import ChannelInterface
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class CachedChannel(ChannelInterface):
+    """
+    CachedChannel wraps an inner channel and caches the data read from it until
+    `num_readers` of readers have read the data. If inner channel is None, the data
+    is written to serialization context and retrieved from there. This is useful
+    when passing data within the same actor and a shared memory channel can be
+    avoided.
+
+    Args:
+        num_readers: The number of readers that will read from this channel. Readers
+            can be the same method of the same actor.
```

Comment:
*must* instead of "can"?

Might also want to specify if this is number of tasks or number of task args, i.e. what happens if the same task reads an arg twice? Can we also add a test for that?

Reply:
Renamed this to `num_reads` and clarified the doc.